utils/platform/os.go

```diff
@@ -179,6 +179,44 @@ func GetRAM() (ram RAM, err error) {
 	return
 }
 
+// SubstituteParameter performs parameter substitution on all platforms.
+// - the first element is the parameter to substitute
+// - if find and replace is also wanted, pass the pattern and the replacement as following arguments in that order.
+func SubstituteParameter(parameter ...string) string {
+	if IsWindows() {
+		return SubstituteParameterWindows(parameter...)
+	}
+	return SubstituteParameterUnix(parameter...)
+}
+
+// SubstituteParameterUnix performs Unix parameter substitution:
+// See https://tldp.org/LDP/abs/html/parameter-substitution.html
+// - the first element is the parameter to substitute
+// - if find and replace is also wanted, pass the pattern and the replacement as following arguments in that order.
+func SubstituteParameterUnix(parameter ...string) string {
+	if len(parameter) < 1 {
+		return "${}"
+	}
+	if len(parameter) < 3 || parameter[1] == "" {
+		return fmt.Sprintf("${%v}", parameter[0])
+	}
+	return fmt.Sprintf("${%v//%v/%v}", parameter[0], parameter[1], parameter[2])
```

Comment:
what if the parameters contain "/", is that possible

Reply:
this is fair enough. I shall add a validation rule

Reply:
Added